File: julia/basics/fizzbuzz.jl
# Print FizzBuzz for 1 through 100, choosing the label from the remainders.
# Run: julia fizzbuzz.jl

function fizzbuzz_label(n::Integer)
    if n % 15 == 0
        "FizzBuzz"
    elseif n % 3 == 0
        "Fizz"
    elseif n % 5 == 0
        "Buzz"
    else
        string(n)
    end
end

for n in 1:100
    println(fizzbuzz_label(n))
end

File: elixir/basics/fizzbuzz.exs
# Print FizzBuzz for 1 through 100, choosing the label by matching a tuple of remainders.
# Run: elixir fizzbuzz.exs

defmodule FizzBuzz do
  def label(n) do
    case {rem(n, 3), rem(n, 5)} do
      {0, 0} -> "FizzBuzz"
      {0, _} -> "Fizz"
      {_, 0} -> "Buzz"
      _ -> Integer.to_string(n)
    end
  end
end

Enum.each(1..100, fn n -> IO.puts(FizzBuzz.label(n)) end)
